docs/static/dead-letter-queues.asciidoc

```diff
@@ -0,0 +1,149 @@
+[[dead-letter-queues]]
+=== Dead Letter Queues
+
+NOTE: The dead letter queue feature is currently supported for the
+<<plugins-outputs-elasticsearch>> output only. Support for additional outputs
+will be available in future releases of the Logstash plugins. Before configuring
+Logstash to use this feature, refer to the output plugin documentation to
+verify that the plugin supports the dead letter queue feature.
+
+By default, when Logstash encounters an event that it cannot process because the
+data contains a mapping error or some other issue, the Logstash pipeline 
+either hangs or drops the unsuccessful event. In order to protect against data
+loss in this situation, you can configure Logstash to write unsuccessful events
+to a dead letter queue instead of dropping them. 
+
+Each event written to the dead letter queue includes the original event along
+with metadata indicating when the event entered the queue. For example:
+
+//TODO: Need a better example here. 
+
+[source,json]
+-------------------------------------------------------------------------------
+{
+          "rand" => "changeme",
+      "sequence" => 9817,
+        "static" => "value",
+    "@timestamp" => 2017-06-06T15:36:48.182Z,
+      "@version" => "1",
+          "host" => "myhost.local"
+}
+-------------------------------------------------------------------------------
+
+To process events in the dead letter queue, you simply create a Logstash
+pipeline configuration that uses the `dead_letter_queue` input plugin
+to read from the queue, process the events, and write to the output. 
+
+image::static/images/dead_letter_queue.png[Diagram showing pipeline reading from the dead letter queue]
+
+See <<processing-dlq-events>> for more information.
+
+
+[[configuring-dlq]]
+==== Configuring Logstash to Use Dead Letter Queues
+
+You enable dead letter queues by setting the `dead_letter_queue_enable` option
+in the  `logstash.yml` <<logstash-settings-file,settings file>>: 
+
+[source,yaml]
+-------------------------------------------------------------------------------
+dead_letter_queue.enable: true
+-------------------------------------------------------------------------------
+
+Dead letter queues are stored as files in the local directory of the Logstash
+instance. By default, the dead letter queue files are stored in
+`path.data/dead_letter_queue`. Each pipeline has a separate queue. For example,
```

Comment:
Mentioning each pipeline has a separate queue only makes sense when we have multi-pipeline. Also, mentioning the "main" pipeline may be confusing to users until multi-pipeline is introduced. I'd suggest massaging this a bit.